src/main/java/net/dv8tion/jda/api/entities/Message.java

```diff
@@ -3035,4 +3039,58 @@ public boolean isSpoiler()
         }
 
     }
+
+    class Interaction implements ISnowflake
+    {
+
+        private final long id;
+        private final int type;
+        private final String name;
+        private final User user;
+        private final Member member;
+
+        public Interaction(long id, int type, String name, User user, Member member)
```

Comment:
I personally think, it's not optimal to have both arguments for user and member.

A user will basically always exist, so it's redundant to pass it when you also pass a non-null member, since you get that said user from the member on creation.

Reply:
Are you suggesting that we have 2 constructors:
`Interaction(long, int, String, User)`
and
`Interaction(long, int, String, Member)` instead?

Reply:
Possible yes, even tho the member / user argument is kind of ambiguous (because you easily could provide a less detailed user instead of a member), but since this is an internal constructor, I probably would go with this it.

Reply:
> per guild avatars and banners are possible or have been teased

They may be in some sort of beta, I've seen them in the wild

Reply:
The current implementation is probably fine.